tests/integrationv2/utils.py

```diff
@@ -49,8 +49,11 @@ def invalid_test_parameters(*args, **kwargs):
 
     if cipher is not None:
         # If the selected protocol doesn't allow the cipher, don't test
-        if protocol is not None and cipher.min_version > protocol:
-            return True
+        if protocol is not None:
+            if cipher.min_version > protocol:
+                return True
+            if protocol is Protocols.TLS13 and cipher.min_version < protocol:
```

Comment:
Nit: this might be easier to reason about if you just do `cipher.min_version < Protocols.TLS13`

Reply:
We only want to do this check if the protocol is TLS13.  We don't want to exclude a TLS1.0 cipher if the protocol is TLS1.2.

However, if the protocol is TLS13, and the cipher is for TLS12, we would have to specify a `-ciphersuites` argument which gives a `no cipher match` error, or specify a `-cipher` argument which gets ignored and a the default TLS13 cipher is used (which fails the test because we aren't expecting a tls13 cipher).